src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -591,6 +626,36 @@ bool StorageDistributedDirectoryMonitor::scheduleAfter(size_t ms)
     return task_handle->scheduleAfter(ms, false);
 }
 
+std::string StorageDistributedDirectoryMonitor::getPath() const
```

Comment:
Ok but... we can return a struct with all data as we now need all this data at once.
(and otherwise we can get inconsisitent data in `system.distribution_queue`)